libs/widget-lib/src/cowSwapWidget.ts

```diff
@@ -149,25 +149,28 @@ function applyDynamicHeight(iframe: HTMLIFrameElement, defaultHeight = DEFAULT_H
 /**
  * Subscribes to the cow events
  *
- * @param events - subscription to different events
+ * @param listeners - subscription to different events
  */
-function subscribeToCoWEvents(events?: CowEventSubscriptions): void {
-  if (!events) {
+function subscribeToCoWEvents(listeners?: CowEventListeners): void {
+  if (!listeners) {
     return
   }
 
+  // Create event emitter
   const eventEmitter = new CowEventEmitterImpl()
 
+  // Subscribe to events
+  for (const listener of listeners) {
+    eventEmitter.on(listener as CowEventListener<CowEvents>)
+  }
+
   window.addEventListener('message', (event) => {
     if (event.data.key !== COW_SWAP_WIDGET_EVENT_KEY || event.data.method !== 'event') {
       return
     }
 
     console.debug(`[CoW event] ${event.data.eventName}`, event.data.payload)
+    eventEmitter
```

Comment:
Hum??

Reply:
This is fixed and move to the `IframeCowEventEmitter` which doesn't have this issue https://github.com/cowprotocol/cowswap/blob/bbb1bcbaae15355060e13ce2032a893ffca2f242/libs/widget-lib/src/IframeCowEventEmitter.ts#L16